test/blas_test.hpp

```diff
@@ -204,19 +204,8 @@ class BLAS_Test<blas_test_args<ScalarT_, MetadataT_, ExecutorType_>>
   }
 };
 
-template <typename T>
-void assert_t_equal(T a, T b) {
-  ASSERT_EQ(a, b);
-}
-
-template <>
-void assert_t_equal<float>(float a, float b) {
-  ASSERT_FLOAT_EQ(a, b);
-}
-
-template <>
-void assert_t_equal<double>(double a, double b) {
-  ASSERT_DOUBLE_EQ(a, b);
-}
+#define ASSERT_T_EQUAL(T, val1, val2)                                         \
+  ASSERT_PRED_FORMAT2(::testing::internal::CmpHelperFloatingPointEQ<T>, val1, \
```

Comment:
You can directly put that in the test instead of wrapping it in a macro.

Reply:
My reasoning for having it a macro is that it allows the code to be less verbose and more clear, and also updatable if a different method is to be called.

"ASSERT_T_EQUAL" is clearer than "ASSERT_PRED_FORMAT2".